File: Anul II/Medii de proiectare si programare/Laborator/lab02/C#/Ubb.BikeContest.Repository/src/IRepository.cs
using System.Collections.ObjectModel;

namespace Ubb.BikeContest.Repository;

public interface IRepository<TID, TEntity> where TEntity : Ubb.BikeContest.Model.Identifiable<TID>
{
    TID Get(TID id);

    Collection<TEntity> Read();

    void Add(TEntity newEntity);

    void Delete(TID id);

    void Update(TEntity updatedEntity);
}

File: Anul II/Medii de proiectare si programare/Laborator/lab03/Java/src/main/java/com/ubb/repository/IRepository.java
package com.ubb.repository;

import com.ubb.model.Identifiable;

import java.util.Collection;
import java.util.Optional;

public interface IRepository<ID, T extends Identifiable<ID>> {
    Optional<T> get(final ID id);

    Collection<T> read();

    void add(final T newEntity);

    void delete(final ID id);

    void update(final T updatedEntity);
}
